pkg/cli/flags.go

```diff
@@ -393,6 +394,10 @@ func init() {
 		// planning?
 		if cmd != connectInitCmd && cmd != connectJoinCmd {
 			cliflagcfg.StringFlag(f, &serverCfg.Attrs, cliflags.Attrs)
+			// Cluster initialization. We only do this for a regular start command;
+			// SQL-only servers get their initialization payload from their tenant
+			// configuration.
+			cliflagcfg.VarFlag(f, configprofiles.NewProfileSetter(&serverCfg.AutoConfigProvider), cliflags.ConfigProfile)
```

Comment:
If I'm reading the code correctly, the server config is given a NoTaskProvider by default and then the NewProfileSetter will install a new provider _if_ the configuration value is set. 👍